Diagnostic/diagnostic.py

```diff
@@ -695,6 +695,26 @@ def start_daemon():
         hutil.error("wait daemon start time out")
 
 
+def check_suspected_memory_leak(pid):
+    memory_leak_threshold_in_KB = 2000000  # Roughly 2GB
+    memory_rss_usage_in_KB = 0
+    memory_leak_suspected = False
+
+    try:
+        # Check /proc/[pid]/status file for "VmRSS" to find out the process's RSS memory usage
+        with open("/proc/{0}/status".format(pid)) as proc_file:
+            for line in proc_file:
+                if line.startswith("VmRSS:"):
+                    memory_rss_usage_in_KB = int(line.split()[1])
+                    memory_leak_suspected = memory_rss_usage_in_KB > memory_leak_threshold_in_KB
+                    break
+    except Exception as e:
+        # Not to throw in case any statement above fails (e.g., invalid pid). Just log.
+        hutil.error("Failed to check memory usage of pid={0}.\nError: {1}\nTrace:\n{2}".format(pid, e, traceback.format_exc()))
+
+    return memory_leak_suspected, memory_rss_usage_in_KB/1024  # return mem usage in MB
```

Comment:
Good point. Let me just return in KB and let the caller do the conversion in logging. Thanks.